### PR TITLE
JAPI-188 Declare file part copy index upon initial dafiserv request

### PR DESCRIPTION
**Pull request by rpastrana**
- Synchronize file part copy index access
- Declare filepartcopy in req json structure

Signed-off-by: Pastrana, Rodrigo <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 1]
```diff
@@ -654,6 +663,9 @@ private String makeNodeObject()
         sb.append("\",\n \"filePart\" : \"");
         sb.append(this.dataPart.getThisPart());
         sb.append("\", \n");
+        sb.append("\"filePartCopy\" : \"");
```
Comment (jpmcmu) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: We only want to specify filePartCopy if we are wanting the copy and not the primary correct?

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 2]
```diff
@@ -114,7 +114,7 @@ public RowServiceInputStream(DataPartition dp, FieldDef rd, FieldDef pRd, int co
         this.readBlock();
     }
 
-    private boolean setNextFilePartCopy()
+    private synchronized boolean setNextFilePartCopy()
```
Comment (jpmcmu) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: Any reason to make this synchronized? Will multiple threads hit this function? I don't see a problem with marking in synchronized, but wanted to make sure I wasn't missing something.

Comment (drealeed) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: Yes, the tardis.hpcc package uses rowServiceInputStream in a multithreaded fashion.

Comment (jpmcmu) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: @drealeed RowServiceInputStream itself isn't thread safe. So each thread should be either be using their own RowServiceInputStream or handling synchronization at a higher level. RowServiceInputStream should be viewed as a typical Java InputStream with no thread safety guarantees

Comment (rpastrana) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: ok agreed, the synchronization at this level is overkill, and might set incorrect expectation

**Comment by rpastrana** [comment]
@jpmcmu , @drealeed  please review

**Comment by jpmcmu** [comment]
@rpastrana looks good

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java (hunk 2 above)
Comment (drealeed) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java]: @jpmcmu good to know, I'll review the hpcc tardis dfu reader and make whatever changes are needed.

**Pull request merged**